File: domain/src/main/java/com/beat/domain/member/repository/MemberRepository.java
package com.beat.domain.member.repository;

import java.util.Optional;

import com.beat.domain.member.domain.Member;
import com.beat.domain.member.domain.SocialType;

public interface MemberRepository {
	Optional<Member> findById(Long id);

	Member save(Member member);

	Optional<Member> findBySocialTypeAndSocialId(Long socialId, SocialType socialType);

	long count();
}

File: domain/src/main/kotlin/com/beat/domain/member/repository/MemberRepository.kt
package com.beat.domain.member.repository

import com.beat.domain.member.model.Member
import com.beat.domain.member.vo.SocialIdentity

interface MemberRepository {
    fun findById(id: Long): Member?

    fun save(member: Member): Member

    fun findBySocialIdentity(socialIdentity: SocialIdentity): Member?

    fun count(): Long
}
